tests/pauli/grouping/test_pauli_group_observables.py

```diff
@@ -309,6 +393,33 @@ def test_return_list_coefficients(self):
         _, grouped_coeffs = group_observables(obs, coeffs)
         assert isinstance(grouped_coeffs[0], list)
 
+    def test_return_new_opmath(self):
+        """Test that using new opmath causes grouped observables to have Prods instead of
+        Tensors"""
+        old_observables = [
+            Tensor(PauliX(0), PauliZ(1)),
+            Tensor(PauliY(2), PauliZ(1)),
+            Tensor(PauliZ(1), PauliZ(2)),
+        ]
+        new_observables = [
+            qml.prod(PauliX(0), PauliZ(1)),
+            qml.prod(PauliY(2), PauliZ(1)),
+            qml.s_prod(1.5, qml.prod(PauliZ(1), PauliZ(2))),
+        ]
+        mixed_observables = [
+            Tensor(PauliX(0), PauliZ(1)),
+            qml.prod(PauliY(2), PauliZ(1)),
+            qml.s_prod(1.5, qml.prod(PauliZ(1), PauliZ(2))),
+        ]
+
+        old_groups = group_observables(old_observables)
+        new_groups = group_observables(new_observables)
+        mixed_groups = group_observables(mixed_observables)
+
+        assert all(isinstance(o, Tensor) for g in old_groups for o in g)
+        assert all(isinstance(o, qml.ops.Prod) for g in new_groups for o in g)
+        assert all(isinstance(o, qml.ops.Prod) for g in mixed_groups for o in g)
```

Comment:
I suppose this is my question - might this cause an issue? this feels somewhat expected to me, but is a little funny

Reply:
The motivation is to avoid complications with e.g. plugins that might still rely on the old system for now

Reply:
I see your point. Again, grouping will work correctly, and to the matter of this raising issues, there should ideally not be any instances where we end up with a list of both `Tensor`s and `Prod`s, and handling that "correctly" shouldn't be expected. Alas, we do not live in an ideal world, so I'm open to discussion if anyone feels strongly about changing this.

Reply:
that sounds ok to me!